thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java

```diff
@@ -217,4 +222,100 @@ public Boolean cloneAnomalyInstances(Long srcId, Long destId) {
     return true;
   }
 
+  /**
+   * Delete raw or merged anomalies whose start time is located in the given time ranges, except
+   * the following two cases:
+   *
+   * 1. If a raw anomaly belongs to a merged anomaly whose start time is not located in the given
+   * time ranges, then the raw anomaly will not be deleted.
+   *
+   * 2. If a raw anomaly belongs to a merged anomaly whose start time is located in the given
+   * time ranges, then it is deleted regardless its start time.
+   *
+   * If monitoringWindowStartTime is not given, then start time is set to 0.
+   * If monitoringWindowEndTime is not given, then end time is set to Long.MAX_VALUE.
+   * @param monitoringWindowStartTime The start time of the monitoring window (in milli-second)
+   * @param monitoringWindowEndTime The start time of the monitoring window (in milli-second)
+   */
+  @POST
+  @Path("function/{id}/deleteExistingAnomalies")
+  public int deleteExistingAnomalies(@PathParam("id") String id,
+      @QueryParam("monitoringWindowStartTime") long monitoringWindowStartTime,
+      @QueryParam("monitoringWindowEndTime") long monitoringWindowEndTime) {
+    LOG.info("Deleting anomalies in the time range: {} -- {}", new DateTime(monitoringWindowStartTime), new
+        DateTime(monitoringWindowEndTime));
+
+    long functionId = Long.valueOf(id);
+
+    int mergedAnomaliesDeleted = 0;
+    AnomalyFunctionDTO anomalyFunction = anomalyFunctionDAO.findById(functionId);
+    LOG.info("Beginning cleanup of functionId {} collection {} metric {}",
+        functionId, anomalyFunction.getCollection(), anomalyFunction.getMetric());
+    // Find merged anomalies and delete them first
+    List<MergedAnomalyResultDTO> mergedResults =
+        mergedAnomalyResultDAO.findByStartTimeInRangeAndFunctionId(monitoringWindowStartTime, monitoringWindowEndTime, functionId);
+    if (CollectionUtils.isNotEmpty(mergedResults)) {
+      mergedAnomaliesDeleted = deleteMergedResults(mergedResults);
+    }
+
+    LOG.info("Deleted {} merged anomalies", mergedAnomaliesDeleted);
+    return mergedAnomaliesDeleted;
+  }
+
+  // Delete merged anomalies from mergedAnomalyResultDAO
+  private int deleteMergedResults(List<MergedAnomalyResultDTO> mergedResults) {
+    LOG.info("Deleting merged results");
+    int mergedAnomaliesDeleted = 0;
+    for (MergedAnomalyResultDTO mergedResult : mergedResults) {
+      // Delete raw anomalies of the merged anomaly
+      List<RawAnomalyResultDTO> rawAnomalyResultDTOs = mergedResult.getAnomalyResults();
+      //deleteRawResults(rawAnomalyResultDTOs);
```

Comment:
In the cloneAnomalyInstances, 
>for (MergedAnomalyResultDTO mergedAnomalyResultDTO : mergedAnomalyResultDTOs) {
>      long oldId = mergedAnomalyResultDTO.getId();
>      mergedAnomalyResultDTO.setId(null);  // clean the Id, then will create a new Id when save
>      mergedAnomalyResultDTO.setRawAnomalyIdList(null);
>      mergedAnomalyResultDTO.setFunctionId(destId);
>      long newId = mergedAnomalyResultDAO.save(mergedAnomalyResultDTO);
>      LOG.debug("clone merged anomaly {} to {}", oldId, newId);
>    }


It doesn't copy the raw anomaly to the new merged results.